lib/client/api.go

```diff
@@ -1001,6 +1001,13 @@ func NewClient(c *Config) (tc *TeleportClient, err error) {
 		tc.HostKeyCallback = tc.localAgent.CheckHostKey
 	}
 
+	// If we haven't already set the private key policy, load it into the client based on the private key.
+	if tc.PrivateKeyPolicy == "" {
+		if key, err := tc.LocalAgent().GetCoreKey(); err == nil {
+			tc.PrivateKeyPolicy = keys.GetPrivateKeyPolicy(key.PrivateKey)
+		}
+	}
```

Comment:
Once https://github.com/gravitational/teleport/pull/20849 is merged (in merge queue), can we merge master and remove this?